runtime/vm/jnimisc.cpp

```diff
@@ -520,6 +520,7 @@ newStringUTF(JNIEnv *env, const char *bytes)
 				gpCheckSetNativeOutOfMemoryError(currentThread, 0, 0);
 				goto done;
 			}
+			data[length*2-1] = '\0';
 			length = encodeUnverifiedUTF8(bytes, length, data);
```

Comment:
The null should be written after the string has been encoded.  I would the `data[length] ='\0';` here

Reply:
will that break anything after this space? you know data[length] is already exceed it's space limit.

Reply:
oh. Do you mean data[length*2] or data[length]? If data[length] , it does not make sense. If data[length*2], I am afraid, it's out of the allocated size.

Reply:
Sorry, I misunderstood your comment.  You are right. We could set data[length]='\0' after the encode. The only concern I have now is if the length returned is exactly same as the memory size. Will it still be safe to set the data[length]='\0'. You know, in that case, this '\0' would be written out fo range.

Reply:
@DanHeidinga  comments? thx.

Reply:
Sorry @zheddie I missed your update in the lead up to the holidays.  

> We could set data[length]='\0' after the encode.

Right, and you're also right about your concern about writing out of range.  To address this, `data` needs to be overallocated by 1 additional U_8 so it's also safe to write there